elks/arch/i86/drivers/block/ata.c

```diff
@@ -0,0 +1,322 @@
+/**********************************************************************
+ * ELKS Generic ATA functions
+ *
+ * This code assumes there is only one ATA controller.
+ *
+ * This code uses LBA addressing for the disks. Any disks without
+ * LBA support (they'd need to be pretty old) are simply ignored.
+ *
+ * All read/write methods check for ERR and DFE (drive-fault err)
+ * There is also an internal timeout counter that once you exceed
+ * ATA_TIMEOUT loops, the current operation is aborted. This is
+ * intended to prevent code hanging on missing or unresponsive ATA
+ * interfaces.
+ *
+ * Caveat emptor: This code is based on the ATA specifications and
+ * some common sense.
+ *
+ * Ferry Hendrikx, June 2025
+ **********************************************************************/
+
+#include <stdint.h>
+#include <linuxmt/config.h>
+#include <linuxmt/kernel.h>
+#include <linuxmt/errno.h>
+#include <linuxmt/debug.h>
+#include <arch/ata.h>
+#include <arch/io.h>
+#include "ata.h"
+
+
+/**********************************************************************
+ * ATA support functions
+ **********************************************************************/
+
+/**
+ * Delays some amount of time (not exactly microseconds)
+ */
+static void delay_us(uint16_t n)
+{
```

Comment:
Let's change this declaration to `unsigned short`, as well as on line 182, and then remove include <stdint.h> above.  At this time, I'd rather keep stdint.h out of kernel includes, and currently the use of unsigned short is quite prevalent within ELKS source and must be 16-bits wide.

Keeping stdint.h would introduce a new standard header file dependency into the kernel build, with the exception of some OS/2 headers files which are shared with external tools.